### PR TITLE
cmake: Switch from tri-state options to boolean. Stage THREE

### PR DESCRIPTION
**Pull request by hebasto**
This PR is a continuation of https://github.com/hebasto/bitcoin/pull/161 and https://github.com/hebasto/bitcoin/pull/162 and tackles with the `WITH_SQLITE` and `WITH_BDB` options.

For the latter the default value `OFF` is suggested.

The https://github.com/hebasto/bitcoin/pull/83#discussion_r1472691247:
> Shouldn't `WITH_SQLITE=ON` imply `ENABLE_WALLET`? What (extra) does `ENABLE_WALLET` do here? I'm not sure we'll actually want to keep it as an option.

has been addressed.

### CODE REVIEW

**Comment by hebasto** [comment]
cc @achow101

**Review by TheCharlatan — APPROVED** [review]
lgtm

File: CMakeLists.txt [hunk 1]
```diff
@@ -67,20 +70,44 @@ option(BUILD_UTIL_CHAINSTATE "Build experimental bitcoin-chainstate executable."
...
+  find_package(BerkeleyDB 4.8 MODULE REQUIRED)
+  set(USE_BDB ON)
+  set(ENABLE_WALLET ON)
+  if(NOT BerkeleyDB_VERSION VERSION_EQUAL 4.8)
+    message(WARNING "Found Berkeley DB (BDB) other than 4.8.\n"
+                    "BDB (legacy) wallets opened by this build will not be portable!"
+    )
+    if(WARN_INCOMPATIBLE_BDB)
+      message(WARNING "If this is intended, pass \"-DWARN_INCOMPATIBLE_BDB=OFF\".\n"
+                      "Passing \"-DWITH_BDB=OFF\" will suppress this warning."
```
[the remark this answers is not in the document]
Comment (hebasto) [reply CMakeLists.txt:96]: It came from https://github.com/hebasto/bitcoin/blob/eb0bdbdd753bca97120247b921fd29d606fea6e9/build-aux/m4/bitcoin_find_bdb48.m4#L48-L66

Any suggestions to improve the behavior?

**Comment by hebasto** [comment]
@TheCharlatan 
> lgtm

Sorry for rebasing. There was a conflict.

Mind approving once more?

**Review by TheCharlatan — APPROVED** [review]
ACK 2b4ad506ee9cd647eacaec8a2df2250fd2b68180

**Pull request merged**